### PR TITLE
feat: settings page - logic, ui

### PR DESCRIPTION
**Pull request by dianafulga**
Solves #46

### CODE REVIEW

**Comment by raducristianpopa** [comment]
<h2>Extension builds preview</h2>

<table role="table">
<thead>
  <tr>
    <th>Name</th>
    <th>Link</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td align="center">Latest commit</td>
    <td align="center">d8c933ada04fcebd9c88ab5ac4de7120f36ecef5</td>
  </tr>
  <tr>
    <td align="center">Latest job logs</td>
    <td align="center"><a href="https://github.com/interledger/web-monetization-extension/actions/runs/7908862839">Run #7908862839</a></td>
  </tr>
  <tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/20768480072/artifacts/1246425343">Download</a></td></tr><tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/20768480072/artifacts/1246425345">Download</a></td></tr><tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/20768480072/artifacts/1246425346">Download</a></td></tr><tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/20768480072/artifacts/1246425344">Download</a></td></tr>
</tbody>
</table>

File: src/popup/pages/Settings.tsx [hunk 1]
```diff
@@ -1,5 +1,151 @@
...
+    <>
+      <Label className="text-base font-medium	text-medium mb-4">Connect your wallet</Label>
+      <div className="mb-4 pl-2">
+        <p className="text-xs font-medium	text-medium inline">
+          Get a wallet address from a provider before connecting it below. Please find a list of
+          available wallets
+        </p>
+        <a href="/" className="text-xs font-medium text-primary inline pl-1">
+          here.
+        </a>
```
Comment (raducristianpopa) [inline src/popup/pages/Settings.tsx:112]:
```suggestion
        <p className="text-xs font-medium	text-medium">
          Get a wallet address from a provider before connecting it below. Please find a list of
          available wallets <a target="_blank" rel="noopener" href="https://webmonetization.org/docs/resources/op-wallets/" className="text-primary">here</a>.
        </p>
```

File: src/popup/pages/Settings.tsx [hunk 2]
```diff
@@ -1,5 +1,151 @@
...
+    setWallet(event.target.value)
+  }
+
+  const handleChangeAmount = (event: any) => {
+    setAmount(event.target.value)
+  }
+
+  const handleChangeRecurring = (value: string) => {
+    setRecurring(value)
+  }
```
Comment (raducristianpopa) [inline src/popup/pages/Settings.tsx:40]: Let's use a boolean type for the `recurring` value.

**Review by ionutanin — APPROVED** [review]
looks good just some small updates based on Radu's feedback

File: src/popup/pages/Settings.tsx [hunk 3]
```diff
@@ -1,5 +1,151 @@
...
+      connected,
+      wmEnabled,
+      wallet,
+      amount,
+      amountType: { recurring },
+    },
+    setData,
+  } = usePopup()
+
+  const [_wallet, setWallet] = useState<string>(wallet)
```
[the remark this answers is not in the document]
Comment (dianafulga) [reply src/popup/pages/Settings.tsx:23]: the underscore is to distinguish from the "wallet" variable that comes from the context provider.

File: src/popup/pages/Settings.tsx (hunk 2 above)
Comment (dianafulga) [reply src/popup/pages/Settings.tsx:40]: input fields work with strings, casting all the time is useless, I would convert at the end when we submit the value

**Pull request merged**